src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -7,14 +7,18 @@
 using System.Linq;
 using System.Net;
 using System.Net.Http.Headers;
+using System.Runtime.InteropServices;
 using Microsoft.Identity.Client.Http;
 using Microsoft.Identity.Client.PlatformsCommon.Shared;
 using Microsoft.Identity.Client.Utils;
+using Microsoft.Win32;
 
 namespace Microsoft.Identity.Client.PlatformsCommon
 {
     internal class DeviceAuthHelper
     {
+        private static bool? s_canOSPerformPKeyAuth;
```

Comment:
maybe use the Lazy<T> initalization as it's thread safe (although this is probably ok as well)